swxsoc/util/tests/test_util.py

```diff
@@ -252,3 +294,131 @@ def test_parse_l0_filenames(filename, instrument, time, level, version, mode):
+    (f"hermes_eea_l1_{time_formatted}_v1.2.3.cdf", "eea", "2024-04-06T12:06:21", "l1", "1.2.3", None),
+    (f"hermes_mrt_l2_{time_formatted}_v1.2.5.cdf", "merit", "2024-04-06T12:06:21", "l2", "1.2.5", None),
+])
+def test_parse_env_var_configured(filename, instrument, time, level, version, mode):
+    """Testing parsing of MOC-generated level 0 files."""
+    # Set SWXSOC_MISSION to 'hermes' mission
+    os.environ["SWXSOC_MISSION"] = "hermes"
+
+    swxsoc._reconfigure()
+    result = util.parse_science_filename(filename)
+    assert result['instrument'] == instrument
+    assert result['level'] == level
+    assert result['version'] == version
+    assert result['time'] == Time(time)
+    assert result['mode'] == mode
+# fmt: on
+
+
+# fmt: off
+@pytest.mark.parametrize("instrument,time,level,version,result", [
+    ("eea", time, "l1", "1.2.3", f"hermes_eea_l1_{time_formatted}_v1.2.3.cdf"),
+    ("merit", time, "l2", "2.4.5", f"hermes_mrt_l2_{time_formatted}_v2.4.5.cdf"),
+    ("nemisis", time, "l2", "1.3.5", f"hermes_nem_l2_{time_formatted}_v1.3.5.cdf"),
+    ("spani", time, "l3", "2.4.5", f"hermes_spn_l3_{time_formatted}_v2.4.5.cdf"),
+]
+)
+def test_create_env_var_configured(instrument, time, level, version, result):
+    """Test simple cases with expected output"""
+    # Set SWXSOC_MISSION to 'hermes' mission
+    os.environ["SWXSOC_MISSION"] = "hermes"
+    # Import the 'util' submodule from 'swxsoc.util'
+    swxsoc._reconfigure()
+    assert (
+        util.create_science_filename(instrument, time, level=level, version=version)
+        == result
+    )
+# fmt: on
+
+
+# fmt: off
+@pytest.mark.parametrize("filename,instrument,time,level,version,mode", [
+    ("mission_INS1_l0_2024094-124603_v01.bin", "instrument1", "2024-04-03T12:46:03", "l0", "01", None),
+    ("mission_INS1_l0_2026337-124603_v11.bin", "instrument1", "2026-12-03T12:46:03", "l0", "11", None),
+    ("mission_INS2_l0_2026215-124603_v21.bin", "instrument2", "2026-08-03T12:46:03", "l0", "21", None),
+    ("mission_INS2_l0_2026337-065422_v11.bin", "instrument2", "2026-12-03T06:54:22", "l0", "11", None),
+    (f"mission_ins1_l1_{time_formatted}_v1.2.3.txt", "instrument1", "2024-04-06T12:06:21", "l1", "1.2.3", None),
+    (f"mission_ins2_l2_{time_formatted}_v1.2.5.txt", "instrument2", "2024-04-06T12:06:21", "l2", "1.2.5", None),
+])
+def test_parse_configdir_configured(filename, instrument, time, level, version, mode):
+    """Testing parsing of MOC-generated level 0 files."""
+    # If the file exists, delete it
+    if os.path.exists(tmp_file_path):
+        os.remove(tmp_file_path)
+
+    # Write the dictionary to a YAML file
+    with open(tmp_file_path, 'w') as file:
+        yaml.dump(config_content, file, default_flow_style=False)
+
+    print(f"Configuration file written to {str(tmp_file_path)}")
+
+    # Set SWXSOC_CONFIGDIR
+    os.environ["SWXSOC_CONFIGDIR"] = str(tmp_file_path.parent)
```

Comment:
How is this env variable used? This might also be good to put in the customization docs.